src/hooks/transfers/skipClient.tsx

```diff
@@ -0,0 +1,55 @@
+import { createContext, useContext, useMemo } from 'react';
+
+import { SkipClient } from '@skip-go/client';
+
+import { getNeutronChainId, getNobleChainId, getOsmosisChainId } from '@/constants/graz';
+import { getSolanaChainId } from '@/constants/solana';
+
+import { getSelectedDydxChainId } from '@/state/appSelectors';
+import { useAppSelector } from '@/state/appTypes';
+
+import { RPCUrlsByChainId } from '@/lib/wagmi';
+
+import { useEndpointsConfig } from '../useEndpointsConfig';
+
+type SkipContextType = ReturnType<typeof useSkipClientContext>;
+const SkipContext = createContext<SkipContextType | undefined>(undefined);
+SkipContext.displayName = 'skipClient';
+
+export const SkipProvider = ({ ...props }) => (
+  <SkipContext.Provider value={useSkipClientContext()} {...props} />
+);
+
+export const useSkipClient = () => useContext(SkipContext);
+
+const useSkipClientContext = () => {
+  const { solanaRpcUrl, nobleValidator, neutronValidator, osmosisValidator, validators } =
+    useEndpointsConfig();
+  const selectedDydxChainId = useAppSelector(getSelectedDydxChainId);
+  const skipClient = useMemo(
+    () =>
+      new SkipClient({
+        endpointOptions: {
+          getRpcEndpointForChain: async (chainId: string) => {
+            if (chainId === getNobleChainId()) return nobleValidator;
+            if (chainId === getNeutronChainId()) return neutronValidator;
+            if (chainId === getOsmosisChainId()) return osmosisValidator;
+            if (chainId === selectedDydxChainId) return validators[0];
```

Comment:
should this be using our `compositClient.network.validatorConfig.restEndpoint` to get the selected validator rather than the first in the list?

Reply:
oh nice!!! thank you! is there a reason why we use `validators[0]` in the [useAccountsBalance hook](https://github.com/dydxprotocol/v4-web/blob/main/src/hooks/useAccountBalance.ts#L142)?

Reply:
Hmmm oh that should be using `compositClient.network.validatorConfig.restEndpoint` as well 😅

Reply:
ooo okay, no worries i can just refactor that in a smol followup PR. just wanted to make sure i understood 🫡